### PR TITLE
audio: Add Cubeb backend

### PR DESCRIPTION
**Pull request by Macdu**
Add a cubeb audio backend to Vita3K.
I tried to do it a bit differently and create a new stream for each Audio out port. 
Because the latency of the PS Vita API and the host may not match, it is still necessary to add some buffering in-between, I tried to minimize.
It is possible to choose in the GUI between the SDL audio backend and the cubeb backend, SDL is chosen by default for now.

The input (microphone) still only uses SDL, it could be implemented using cubeb later.

### CODE REVIEW

File: vita3k/gui/src/settings_dialog.cpp [hunk 1]
```diff
@@ -1029,8 +1045,7 @@ void draw_settings_dialog(GuiState &gui, EmuEnvState &emuenv) {
     const auto is_reboot = (emuenv.renderer->current_backend != emuenv.backend_renderer) || (config.resolution_multiplier != emuenv.cfg.current_config.resolution_multiplier);
     if (ImGui::Button(is_apply ? (is_reboot ? "Save & Reboot" : "Save & Apply") : "Save", BUTTON_SIZE)) {
         save_config(gui, emuenv);
-        if (is_apply)
-            set_config(gui, emuenv, emuenv.io.app_path);
+        set_config(gui, emuenv, emuenv.io.app_path);
```
Comment (Zangetsu38) [inline vita3k/gui/src/settings_dialog.cpp:1048]: no touch on this please ^^

File: vita3k/gui/src/settings_dialog.cpp [hunk 2]
```diff
@@ -665,6 +667,20 @@ void draw_settings_dialog(GuiState &gui, EmuEnvState &emuenv) {
         ImGui::Spacing();
         ImGui::Checkbox("Boot apps in full screen", &emuenv.cfg.boot_apps_full_screen);
         ImGui::Spacing();
+
+        if (!emuenv.io.app_path.empty())
+            ImGui::BeginDisabled();
+
+        static const char *LIST_BACKEND_AUDIO[] = { "SDL", "Cubeb" };
+        static int audio_backend_idx = (emuenv.audio.audio_backend == "SDL") ? 0 : 1;
```
Comment (Zangetsu38) [inline vita3k/gui/src/settings_dialog.cpp:675]: move this inside init setting function, like other backend value

File: external/CMakeLists.txt [hunk 3]
```diff
@@ -117,6 +117,28 @@ add_library(sdl2 INTERFACE)
...
+	target_compile_definitions(speex PUBLIC
+		#_USE_NEON
+	)
+elseif (CMAKE_SYSTEM_PROCESSOR MATCHES "^(x86|X86|amd64|AMD64|em64t|EM64T)")
+	target_compile_definitions(speex PUBLIC
+		_USE_SSE
+		_USE_SSE2
+	)
+endif ()
+
```
Comment (bookmist) [inline external/CMakeLists.txt:141]: also add 
```set_property(TARGET cubeb PROPERTY FOLDER externals)
set_property(TARGET speex PROPERTY FOLDER externals)
```
to put external projects into external folder in msvc

File: vita3k/gui/src/settings_dialog.cpp [hunk 4]
```diff
@@ -202,6 +203,7 @@ static CPUBackend set_cpu_backend(std::string &cpu_backend) {
 }
 
 static int current_aniso_filter_log, max_aniso_filter_log;
+static int audio_backend_idx;
```
Comment (Zangetsu38) [inline vita3k/gui/src/settings_dialog.cpp:206]: no need news line here ^^
```suggestion
static int audio_backend_idx, current_aniso_filter_log, max_aniso_filter_log;
```

**Review by Zangetsu38 — APPROVED** [review]

**Comment by Zangetsu38** [comment]
Already merged

**Pull request closed without merging**